src/main/java/com/datasonnet/plugins/JavaFormatPlugin.java

```diff
@@ -0,0 +1,96 @@
+package com.datasonnet.plugins;
+
+import com.datasonnet.document.Document;
+import com.datasonnet.document.JavaObjectDocument;
+import com.datasonnet.document.StringDocument;
+import com.datasonnet.spi.DataFormatPlugin;
+import com.datasonnet.spi.PluginException;
+import com.datasonnet.spi.UjsonUtil;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JavaType;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import ujson.Value;
+
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class JavaFormatPlugin implements DataFormatPlugin {
+    public static String OUTPUT_CLASS = "OutputClass";
+    public static String DATE_FORMAT = "DateFormat";
+
+    public static String DEFAULT_DATE_FORMAT = "yyyy-MM-dd'T'HH:mm:ss.SSSZ";
+
+    public JavaFormatPlugin() { }
+
+    public Value read(Object input, Map<String, Object> params) throws PluginException {
+        ObjectMapper mapper = new ObjectMapper();
+        DateFormat df = new SimpleDateFormat(params.containsKey(DATE_FORMAT) ? (String)params.get(DATE_FORMAT) : DEFAULT_DATE_FORMAT);
```

Comment:
This is almost certainly an area we'll want extensive documentation on behavior under different circumstances